Added Support for TSQL's constraint_table_usage

oscarno22 opened the pull request:
### Description

Currently, Babelfish does not support
information_schema_tsql.constraint_table_usage. This commit updates this
state by implementing the view.

TASK:

Signed-off-by: Ozzy Nolen <oscarno@amazon.com>
 
### Issues Resolved

- The view information_schema_tsql.constraint_table_usage not implemented


### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

oscarno22 committed:
Added Support for TSQL's constraint_table_usage

Currently, Babelfish does not support
information_schema_tsql.constraint_table_usage. This commit updates this
state by implementing the view.

TASK:

Signed-off-by: Ozzy Nolen <oscarno@amazon.com>

shalinilohia50 commented on test/JDBC/upgrade/latest/schedule:
```diff
@@ -149,3 +149,4 @@ BABEL-3213
 BABEL-RAND
 BABEL-741
 BABEL-ROLE-MEMBER
+constraint_table_usage
```
Add this test in the schedule files of all the upgrades.

shalinilohia50 commented on test/JDBC/input/constraint_table_usage.sql:
```diff
@@ -0,0 +1,51 @@
+use master;
```
We can remove this test file as we already have tests in prepare and verify format. This just makes a redundant test file.

shalinilohia50 commented on test/JDBC/input/constraint_table_usage-vu-prepare.sql:
```diff
@@ -0,0 +1,26 @@
+use master;
+go
+
+drop database if exists constraint_table_usage_db;
```
No need to drop the database `constraint_table_usage_db` here.

shalinilohia50 commented on test/JDBC/input/constraint_table_usage-vu-prepare.sql:
```diff
@@ -0,0 +1,26 @@
+drop database if exists constraint_table_usage_db;
+go
+
+create database constraint_table_usage_prepare_db;
```
keep the names as `constraint_table_usage_vu_prepare_db`

shalinilohia50 commented on test/JDBC/expected/constraint_table_usage-vu-verify.out:
```diff
@@ -0,0 +1,48 @@
+use constraint_table_usage_prepare_db;
+go
+
+select * from information_schema.constraint_table_usage where table_name like 'constraint_table_usage_tb%';
```
The table names are starting with `constraint_table_usage_prepare_tb%`. Please modify it.

HarshLunagariya committed:
Fix DEFINITION column in information_schema.views

Previously, DEFINITION column was returning NULL when length was more
than 4000. This commit fixes DEFINITION column such that it returns
truncated definition when length is more than 4000.

Task: BABEL-2889
Signed-off-by: Harsh Lunagariya <lunharsh@amazon.com>

shalinilohia50 committed:
Missing privilege on the constraint_column_usage view

The user that can access a table should be able to see its data in the view.
Added tests to verify this behaviour.

Task: BABEL-3439
Signed-off-by: Shalini Lohia lshalini@amazon.com

bsatarupa committed:
Numeric testcase for precision overflow triggers known protocol violation

Before Fix:
Numeric testcase for precision overflow triggers known protocol violation

After Fix:
Postgres can store numeric data upto 1000 digits but SQL server has a maximum limit of 38 digits. Since, adding the maximum numeric precision check on TDS sender side results into protocol violation, have addressed the issue on engine side by calculating exact numeric digits from Postgres numeric-blocks format and reporting error whenever #digits > TDS_MAX_NUMERIC_PRECISION(i.e. 38).

Task: BABEL-3179
Authored-by: Satarupa Biswas satarupb@amazon.com
Signed-off-by: Kuntal Ghosh <kuntalgh@amazon.com>

oscarno22 committed:
Updated Testing Files

This commit adjusts naming practices for objects created in test. It
also removes constraint_table_usage.sql, as vu-prepare and vu-verify are
adequate. Finally, this commit adds constraint_table_usage to the
schedule file of all of the upgrades.

TASK:

Signed-off-by: Ozzy Nolen <oscarno@amazon.com>
